### ISSUE TITLE
Feature request: create namespace alerts via API calls 

### ISSUE THREAD

**Issue by pierluigilenoci** [opencost/opencost#766]
We would like to define `Namespace Alerts` via API calls.

**Comment by dwbrown2** [comment]
@pierluigilenoci any more details you can share on what would be helpful? Most teams have requested doing this via YAML, would love to hear what would be helpful for you.

**Comment by pierluigilenoci** [comment]
@dwbrown2 we currently manage our namespaces by separating them by environments and projects. We have about a hundred different projects and an average of 4 environments per project, a total of more than 400 different namespaces. Managing alerts for 400 namespaces via a YAML file doesn't make any sense.

From our point of view, the perfect solution is an API Rest with which to create/modify/delete alerts by namespace (or possibly other alerts by project/department, etc ref: https://github.com/kubecost/cost-model/issues/765) with automation (powered by our internal governance application).

**Comment by dwbrown2** [comment]
We currently have an API for this, but it's undergoing some changes as move from the `checks` jobs to our core Go application. We can share documentation once it's ready if that would be helpful.

**Comment by pierluigilenoci** [comment]
@dwbrown2 I look forward to the release of the new version.

**Comment by dwbrown2** [comment]
@pierluigilenoci there are two relevant APIs here. Would be interested in your general feedback/questions and then we can look to document further!

`<your-kubecost-endpoint>/api/setNamespaceAttributes` -- used to set contact info

Post Parameters
```
{"namespaces":
  {"data-science":  // namespace name
    {"ownerLabel":"Tester","email":"test@example.com","sendAlert":"true"}
  }
}
```
Response: JSON representation of namespace metadata

`<your-kubecost-endpoint>/api/setBudget` -- used to set namespace budget alert

Post Parameters
```
{
  "value":"20", // daily budget value
  "alert":"true", // alerts enabled or disabled
  "namespace":"data-science"  // namespace name
} 


Response: JSON representation of budget

**Comment by pierluigilenoci** [comment]
This is exactly what I was looking for. ❤️ 

I've tried both APIs and there are some things I don't understand.

**setBudget**

When I call `setBudget` I get as a result the list of budgets of all namespaces but in the head, there is an empty JSON. 
```json
{
    "byNamespace": {
        "": {
            "value": "",
            "alert": "",
            "namespace": ""
        },
		"OTHER NAMESPACES JSON": "REDACTED"
    }
}
``` 

Is this normal?

**setNamespaceAttributes**

If I want to configure all the namespaces I have to send a payload with all the namespaces.
If one is not listed the previous configuration is deleted.

Is this a deliberate behavior? 

**In conclusion**

These behavior seem child of the fact that these APIs are probably used by the Kubecost UI which has all the namespaces listed on a page and the budget managed by namespace detail page. It's a little bit counterintuitive for those who use alone the API.

Apart from the doubts above if there is also detailed documentation on how to implement the alerts with the API I would say that the case is closed.

**Comment by mattray** [comment]
This issue is being closed because it may not be relevant to the OpenCost project and appears stale. If you feel this was closed in error, please open a [new OpenCost issue](https://github.com/opencost/opencost/issues) with updated details or if it is still relevant for Kubecost please open an issue with [Kubecost Support](https://support.kubecost.com).
